### DISCUSSION TITLE
Does the library support C++20 std::views::drop?

### DISCUSSION THREAD

**Discussion by Jack012a** [nlohmann/json#3562] [Q&A] [answered]
I tried the following and it doesn't work for me.

```
    std::string str = R"([1, 2, 3, 4, 5])";
    json j_object = json::parse(str);
    for (const auto &it : j_object | std::views::drop(1))
    {
        std::cout << it << std::endl;
    }
```

json.hpp:12059: bool nlohmann::detail::iter_impl<BasicJsonType>::operator==(const IterImpl&) const [with IterImpl = nlohmann::detail::iter_impl<nlohmann::basic_json<> >; typename std::enable_if<(std::is_same<IterImpl, nlohmann::detail::iter_impl<BasicJsonType> >::value || std::is_same<IterImpl, nlohmann::detail::iter_impl<typename std::conditional<std::is_const<_Tp>::value, typename std::remove_const<_Tp>::type, const BasicJsonType>::type> >::value), std::nullptr_t>::type <anonymous> = nullptr; BasicJsonType = nlohmann::basic_json<>]: Assertion `m_object != nullptr' failed.

**Comment by falbrechtskirchinger** [comment, marked as answer]
Please test with `develop`. This is expected to work now (and in the upcoming 3.11.0 release).

**Reply by nlohmann** [reply]
@Jack012a Do you need further assistance?
